### PR TITLE
feat: add Job constarint

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -570,6 +570,8 @@ components:
           type: string
         productType:
           type: string
+        extraConstraintData:
```
Comment (cl-shaharshavit-rnd) [inline openapi3.yaml:573]: additionalIdentifiers

is better name

File: src/DAL/entity/job.ts [hunk 2]
```diff
@@ -67,6 +68,9 @@ export class JobEntity {
   @Column('text', { nullable: true })
   public productType: string;
 
+  @Column('text', { default: '', nullable: false })
+  public extraConstraintData: string;
```
Comment (cl-shaharshavit-rnd) [inline src/DAL/entity/job.ts:72]: additionalIdentifiers

is better name

File: src/DAL/migration/FullSchema.sql [hunk 3]
```diff
@@ -33,8 +33,9 @@ CREATE TABLE public."Job"
   "expiredTasks" int NOT NULL DEFAULT 0,
   "pendingTasks" int NOT NULL DEFAULT 0,
   "inProgressTasks" int NOT NULL DEFAULT 0,
+  "extraConstraintData" text COLLATE pg_catalog."default" NOT NULL DEFAULT ''::text,
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply src/DAL/migration/FullSchema.sql:36]: why not null?

File: src/DAL/migration/v3.1.3.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE public."Job"
+  ADD COLUMN "extraConstraintData" text COLLATE pg_catalog."default" NOT NULL DEFAULT ''::text,
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply src/DAL/migration/v3.1.3.sql:2]: why not null?

File: src/common/dataModels/jobs.ts [hunk 5]
```diff
@@ -77,6 +78,7 @@ export interface IGetJobResponse {
   expiredTasks: number;
   pendingTasks: number;
   inProgressTasks: number;
+  extraConstraintData?: string;
```
Comment (cl-shaharshavit-rnd) [inline src/common/dataModels/jobs.ts:81]: additionalIdentifiers

is better name

File: tests/unit/DAL/jobModelConverter.spec.ts [hunk 6]
```diff
@@ -162,6 +162,7 @@ describe('JobModelConverter', function () {
         producerName: 'test producer',
         productType: 'test type',
         productName: 'test product',
+        extraConstraintData: '',
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/DAL/jobModelConverter.spec.ts:165]: additionalIdentifiers

is better name

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: src/DAL/entity/job.ts [hunk 7]
```diff
@@ -68,8 +68,8 @@ export class JobEntity {
   @Column('text', { nullable: true })
   public productType: string;
 
-  @Column('text', { default: '', nullable: false })
-  public extraConstraintData: string;
+  @Column('text', { nullable: true })
+  public additionalIdentifiers: string;
```
Comment (cl-shaharshavit-rnd) [inline src/DAL/entity/job.ts:72]: the typescript model is still not nullable.....

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

**Review by cl-shaharshavit-rnd — APPROVED** [review]

**Review by AnguIar — APPROVED** [review]

**Pull request merged**